src/app/services/auth.service.ts

```diff
@@ -25,14 +32,15 @@ export class AuthService {
     authStateChange(state) {
       this.authState = state;
       this.authStateSource.next(this.authState);
+      this.isAuth = this.authState.isLoggedIn;
     }
 
     /**
      * To affirm that user is logged in
      * @param autoFetchUserDetails  User details fetch flag
      */
     loggedIn(autoFetchUserDetails = false) {
-      this.authState = {isLoggedIn: true};
+      this.authStateChange({isLoggedIn: true, username: ''});
```

Comment:
@sanketbansal Why `username: ''`?

Reply:
Because authState is binded with the UI. And whenever there is a change in authState UI gets updated and if the username is **null**  UI crashes.

Reply:
fetchUserDetails method updates authState with proper username after that.